packages/server/src/fhir/search.ts

```diff
@@ -526,42 +549,165 @@ async function getSearchRevIncludeEntries(
+    if (canUseCursorLinks(searchRequest)) {
+      buildSearchLinksWithCursor(searchRequest, entries, hasMore, result);
+    } else {
+      buildSearchLinksWithOffset(searchRequest, hasMore, result);
+    }
+  }
+
+  return result;
+}
 
+/**
+ * Returns true if the search request can use cursor links.
+ * Cursor links are more efficient than offset links for large result sets.
+ * A search request can use cursor links if:
+ *   1. Not using offset pagination
+ *   2. Exactly one sort rule using _lastUpdated ascending
+ * @param searchRequest - The candidate search request.
+ * @returns True if the search request can use cursor links.
+ */
+function canUseCursorLinks(searchRequest: SearchRequestWithCountAndOffset): boolean {
+  return (
+    searchRequest.offset === 0 &&
+    searchRequest.sortRules?.length === 1 &&
+    searchRequest.sortRules[0].code === '_lastUpdated' &&
+    !searchRequest.sortRules[0].descending
+  );
+}
+
+/**
+ * Builds the "first", "next", and "previous" links for a search request using cursor pagination.
+ * @param searchRequest - The search request.
+ * @param entries - The search bundle entries.
+ * @param hasMore - True if there are more entries after the current page.
+ * @param result - The search bundle links.
+ */
+function buildSearchLinksWithCursor(
+  searchRequest: SearchRequestWithCountAndOffset,
+  entries: BundleEntry[],
+  hasMore: boolean | undefined,
+  result: BundleLink[]
+): void {
+  let currCursor: Cursor | undefined = undefined;
+  if (searchRequest.cursor) {
+    currCursor = parseCursor(searchRequest.cursor);
+  }
+  if (!currCursor) {
+    currCursor = {
+      version: '1',
+      instant: '',
+    };
+  }
+
+  result.push({
+    relation: 'first',
+    url: getSearchUrl({ ...searchRequest, cursor: undefined, offset: undefined }),
+  });
+
+  if (hasMore) {
     result.push({
-      relation: 'first',
-      url: getSearchUrl({ ...searchRequest, offset: 0 }),
+      relation: 'next',
+      url: getSearchUrl({
+        ...searchRequest,
+        cursor: formatCursor({ ...currCursor, instant: getMaxLastUpdated(entries) }),
+        offset: undefined,
+      }),
     });
+  }
+}
 
-    if (hasMore) {
-      result.push({
-        relation: 'next',
-        url: getSearchUrl({ ...searchRequest, offset: offset + count }),
-      });
-    }
+/**
+ * Parses a cursor string into a Cursor object.
+ * @param cursor - The cursor string.
+ * @returns The Cursor object or undefined if the cursor string is invalid.
+ */
+function parseCursor(cursor: string): Cursor | undefined {
+  const parts = splitN(cursor, '-', 2);
+  if (parts.length !== 2) {
+    return undefined;
+  }
+  return { version: parts[0], instant: parts[1] };
+}
+
+/**
+ * Formats a cursor object into a cursor string.
+ * @param cursor - The cursor object.
+ * @returns The cursor string.
+ */
+function formatCursor(cursor: Cursor): string {
+  return `${cursor.version}-${cursor.instant}`;
```

Comment:
Yeah, you mentioned that yesterday, and I looked into it.  Perhaps I am rationalizing laziness, but here's my reasons for not doing it:

1. General YAGNI
2. Base64 will add friction, but ultimately nothing will truly prevent people if they want to reverse engineer
3. In this case, there really isn't any advantage to reverse engineering, because this is just a convenience around `_lastUpdated` filters
4. There's also a performance consideration that we use `repo.search()` all over the place internally, and often do not need the `Bundle.link` values.  I know base64 encoding will be fast and cheap relative to the actual db query, but it still feels wrong.  That could probably be avoided with some kind of `_nolinks=true` option, but that's additional work.